lib/android.dart

```diff
@@ -269,28 +272,35 @@ List<_AndroidDrawableTemplate> _getAssociatedTemplates({
 /// Note: Do not change interpolation unless you end up with better results
 /// https://github.com/fluttercommunity/flutter_launcher_icons/issues/101#issuecomment-495528733
 void _saveImageAndroid({
-  required _AndroidDrawableTemplate template,
+  required List<_AndroidDrawableTemplate> templates,
   required Image image,
   required fileName,
-}) {
-  //added file name attribute to make this method generic for splash image and branding image.
-  final newFile = copyResize(
-    image,
-    width: image.width * template.pixelDensity ~/ 4,
-    height: image.height * template.pixelDensity ~/ 4,
-    interpolation: Interpolation.cubic,
-  );
-
-  // Whne the flavor value is not specified we will place all the data inside the main directory.
-  // However if the flavor value is specified, we need to place the data in the correct directory.
-  // Default: android/app/src/main/res/
-  // With a flavor: android/app/src/[flavor name]/res/
-  final file = File(
-    '${_flavorHelper.androidResFolder}${template.directoryName}/$fileName',
+  required String androidResFolder,
```

Comment:
I don't understand why you are passing this variable that is already accessible to the function?

Reply:
Hey it has to do with how Isolates work with global variables. If I'm not mistaken, when you spin up another Isolate it has access to global variables but since the `_flavorHelper` is a late property, the new Isolate simply doesn't have it initialised in it's memory pool. That's why I had to pass the value from the initial Isolate to the new one being created at this point.
How does that sound to you?